Disable tagged names

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26366 Refactor Dimname.h API to be nicer
* #26365 Delete tagged names
* **#26356 Disable tagged names**

This PR doesn't delete the code for them yet because it takes some effort to
determine what to delete. I will send a followup PR fully deleting
tagged names, but this PR disables their creation.

Test Plan:
- [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/26356

Differential Revision: [D17428409](https://our.internmc.facebook.com/intern/diff/D17428409)

zou3519 committed:
Disable tagged names

This PR doesn't delete the code for them yet because it takes some effort to
determine what to delete. I will send a followup PR fully deleting
tagged names, but this PR disables their creation.

Test Plan:
- [namedtensor ci]

zou3519 committed:
Update on "Disable tagged names"


This PR doesn't delete the code for them yet because it takes some effort to
determine what to delete. I will send a followup PR fully deleting
tagged names, but this PR disables their creation.

Test Plan:
- [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/26356

Differential Revision: [D17428409](https://our.internmc.facebook.com/intern/diff/D17428409)

nairbv commented on aten/src/ATen/core/Dimname.cpp:
```diff
@@ -54,26 +54,8 @@ Dimname Dimname::fromSymbol(Symbol full_name) {
-  check_valid_identifier(tag);
-  return Dimname(NameType::TAGGED, full_name, Symbol::dimname(untagged_name));
+  check_valid_identifier(full_name.toUnqualString());
+  return Dimname(full_name);
```
if we're getting rid of tagged names, should it just be "name" instead of "full_name"?

zou3519 replied:
I changed that in the next diff in the stack: https://github.com/pytorch/pytorch/pull/26365.

This one is to just disable creation of tagged names.

zou3519 replied:
I can merge the two PRs together (perhaps they make more sense as one unit) if that makes them easier to review.

nairbv replied:
yup, saw it.

nairbv commented on aten/src/ATen/test/Dimname_test.cpp:
```diff
@@ -31,32 +31,17 @@ TEST(DimnameTest, wildcardName) {
   ASSERT_EQ(dimname.full_name(), foo);
-  ASSERT_EQ(dimname.untagged_name(), foo);
-
+  ASSERT_THROW(Dimname::fromSymbol(Symbol::dimname("inva.lid")), c10::Error);
```
what are we throwing in this case? that `.` isn't allowed?

zou3519 replied:
We are throwing the error message that says "only alphabetical and underscore characters are allowed"

zou3519 committed:
Update on "Disable tagged names"


This PR doesn't delete the code for them yet because it takes some effort to
determine what to delete. I will send a followup PR fully deleting
tagged names, but this PR disables their creation.

Test Plan:
- [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/26356

Differential Revision: [D17428409](https://our.internmc.facebook.com/intern/diff/D17428409)

zou3519 committed:
Update on "Disable tagged names"


This PR doesn't delete the code for them yet because it takes some effort to
determine what to delete. I will send a followup PR fully deleting
tagged names, but this PR disables their creation.

Test Plan:
- [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/26356

Differential Revision: [D17428409](https://our.internmc.facebook.com/intern/diff/D17428409)
